app/src/main/java/com/movies/domain/usecase/movies/GetGenresUseCase.kt

```diff
@@ -0,0 +1,14 @@
+package com.movies.domain.usecase.movies
+
+import com.movies.domain.model.GenreDomainModel
+import com.movies.domain.repository.MoviesRepository
+import com.movies.domain.usecase.base.BaseUseCase
+
+class GetGenresUseCase(
+    private val moviesRepository: MoviesRepository
+) : BaseUseCase<Unit, List<GenreDomainModel>>() {
+
+    override suspend fun invoke(params: Unit?): List<GenreDomainModel> {
+        return moviesRepository.fetchMovieGenre()
+    }
```

Comment:
i think, its better to name usecase same name as a function. FetchMovieGenreUseCase

Reply:
did it in last PR to prevent conflicts from merging several branches into one another